cmd/goal/tealsign.go

```diff
@@ -0,0 +1,223 @@
+// Copyright (C) 2019-2020 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"encoding/base32"
+	"encoding/base64"
+	"io/ioutil"
+
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/data/transactions/logic"
+	"github.com/algorand/go-algorand/protocol"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	keyFilename     string
+	signerAcct      string
+	lsigTxnFilename string
+	contractAddr    string
+	signTxID        bool
+	dataFile        string
+	datab64         string
+	datab32         string
+	setLsigArg      int
+)
+
+func init() {
+	clerkCmd.AddCommand(tealsignCmd)
+
+	tealsignCmd.Flags().StringVar(&keyFilename, "keyfile", "", "algokey private key file to sign with")
+	tealsignCmd.Flags().StringVar(&signerAcct, "account", "", "address of account to sign with")
+	tealsignCmd.Flags().StringVar(&lsigTxnFilename, "lsig-txn", "", "transaction with logicsig to sign data for")
+	tealsignCmd.Flags().StringVar(&contractAddr, "contract-addr", "", "contract address to sign data for. not necessary if --lsig-txn is provided")
+	tealsignCmd.Flags().BoolVar(&signTxID, "sign-txid", false, "use the txid of --lsig-txn as the data to sign")
+	tealsignCmd.Flags().StringVar(&dataFile, "data-file", "", "data file to sign")
+	tealsignCmd.Flags().StringVar(&datab64, "data-b64", "", "base64 data to sign")
+	tealsignCmd.Flags().StringVar(&datab32, "data-b32", "", "base32 data to sign")
+	tealsignCmd.Flags().IntVar(&setLsigArg, "set-lsig-arg", -1, "if --lsig-txn is also specified, set this lsig arg to the raw signature bytes")
```

Comment:
Add a note saying that if this is provided, then in `lsigTxnFilename` will be overwritten such that the lsig arg at the position here will be overwritten with the signature.

Reply:
It might be a good idea to rename this to `set-lsig-arg-index`

Might also want to note that `nil` lsig args will be appended until the index is valid.

Reply:
done, thanks!